TOML encoding C compilation error

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.10, press to see full `v doctor` output</summary>

|V full version      |V 0.4.10 9b1937a87166e3327497f332bf9584ff90592617.
|:-------------------|:-------------------
|OS                  |macos, macOS, 15.4.1, 24E263
|Processor           |8 cpus, 64bit, little endian, Apple M1 Pro
|Memory              |0.24GB/16GB
|                    |
|V executable        |/opt/homebrew/Cellar/vlang/0.4.10/libexec/v
|V last modified time|2025-03-20 11:42:00
|                    |
|V home dir          |OK, value: /opt/homebrew/Cellar/vlang/0.4.10/libexec
|VMODULES            |OK, value: /Users/steve/.vmodules
|VTMP                |OK, value: /tmp/v_501
|Current working dir |OK, value: /Users/steve/code/github.com/stilvoid/bleng
|                    |
|Git version         |git version 2.39.5 (Apple Git-154)
|V git status        |4.5.1 (64816 commit(s) behind V master)
|.git/config present |false
|                    |
|cc version          |Apple clang version 17.0.0 (clang-1700.0.13.3)
|gcc version         |Apple clang version 17.0.0 (clang-1700.0.13.3)
|clang version       |Apple clang version 17.0.0 (clang-1700.0.13.3)
|tcc version         |N/A
|tcc git status      |N/A
|emcc version        |N/A
|glibc version       |N/A

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  toml_bug.v && toml_bug`
```v
module main

import toml

fn main() {
    // Create a TOML document
    mut doc := map[string]toml.Any{}
    doc['key'] = toml.Any('value')
    
    // Try to encode it - this is where the bug occurs
    toml_content := toml.encode(doc)
    
    println(toml_content)
}

```

**What did you see?**
```
================== C compilation error (from cc): ==============
cc: /tmp/v_501/toml_bug.01JTRA3CD8YK3RM5YZNS5PM84Q.tmp.c:12313:12: error: unexpected type name 'map': expected expression
cc:  12313 |                                 return map[string]toml__Any_to_toml(typ);
cc:        |                                        ^
cc: /tmp/v_501/toml_bug.01JTRA3CD8YK3RM5YZNS5PM84Q.tmp.c:12313:16: error: unexpected type name 'string': expected expression
cc:  12313 |                                 return map[string]toml__Any_to_toml(typ);
cc:        |                                            ^
cc: /tmp/v_501/toml_bug.01JTRA3CD8YK3RM5YZNS5PM84Q.tmp.c:12313:23: error: expected ';' after return statement
cc:  12313 |                                 return map[string]toml__Any_to_toml(typ);
cc:        |                                                   ^
cc:        |                                                   ;
cc: 3 errors generated.
================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

Correctly encoded toml or a V error, not a C compilation error.


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/toml/toml.v
+++ b/vlib/toml/toml.v
@@ -167,13 +167,18 @@ fn decode_struct[T](doc Any, mut typ T) {
 // encode encodes the type `T` into a TOML string.
 // If `T` has a custom `.to_toml()` method, it will be used instead of the default.
 pub fn encode[T](typ T) string {
-	$for method in T.methods {
-		$if method.name == 'to_toml' {
-			return typ.$method()
+	$if T is $struct {
+		$for method in T.methods {
+			$if method.name == 'to_toml' {
+				return typ.$method()
+			}
 		}
+		mp := encode_struct[T](typ)
+		return mp.to_toml()
+	} $else {
+		$compile_error('Currently only type `struct` is supported for `T` to encode as TOML')
 	}
-	mp := encode_struct[T](typ)
-	return mp.to_toml()
+	return ''
 }
 
 fn encode_struct[T](typ T) map[string]Any {

